lib/matplotlib/animation.py

```diff
@@ -1156,8 +1160,9 @@ def _init_draw(self):
             self._draw_frame(next(self.new_frame_seq()))
         else:
             self._drawn_artists = self._init_func()
-            for a in self._drawn_artists:
-                a.set_animated(self._blit)
+            if iterable(self._drawn_artists):
```

Comment:
If you are using blitting and return `None` blitting is effectively turned off (due to the logic it `_post_draw`) and if `self._drawn_artists is not an iterable the code in `_blit_draw` would raise.